### ISSUE TITLE
Package won't install with ocaml 4.06.1 on mac os x 

### ISSUE THREAD

**Issue by groar** [dbousque/lymp#13]
I tried to install lymp with `ocaml-4.06.1` on mac OS X, and it seems that the required dependency `bson` won't build using this version of ocaml. There is no problem with the `4.05` version though.

```$ opam install lymp
The following actions will be performed:
  ∗  install bson 0.89.3                      [required by lymp]
  ∗  install lymp 0.2.3
===== ∗  2 =====
Do you want to continue ? [Y/n] y

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫
[bson] Archive in cache
[lymp] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫
[ERROR] The compilation of bson failed at "ocaml setup.ml -configure --disable-syntax --disable-client --prefix /Users/xxx/.opam/4.06.1".
Processing  1/2: [bson: ocamlfind remove]
#=== ERROR while installing bson.0.89.3 =======================================#
# opam-version 1.2.2
# os           darwin
# command      ocaml setup.ml -configure --disable-syntax --disable-client --prefix /Users/xxx/.opam/4.06.1
# path         /Users/xxx/.opam/4.06.1/build/bson.0.89.3
# compiler     4.06.1
# exit-code    2
# env-file     /Users/xxx/.opam/4.06.1/build/bson.0.89.3/bson-47809-d2c37b.env
# stdout-file  /Users/xxx/.opam/4.06.1/build/bson.0.89.3/bson-47809-d2c37b.out
# stderr-file  /Users/xxx/.opam/4.06.1/build/bson.0.89.3/bson-47809-d2c37b.err
### stderr ###
# File "./setup.ml", line 247, characters 8-26:
# Warning 3: deprecated: String.set
# Use Bytes.set instead.
# File "./setup.ml", line 247, characters 8-11:
# Error: This expression has type string but an expression was expected of type
#          bytes

```

**Comment by groar** [comment]
Just to check, I also tried it with the `ocaml/opam:ubuntu-16.04_ocaml-4.06.0` docker image, and the same problem arises as well (as expected).

**Comment by dbousque** [comment]
Thank you very much for the report. Indeed, `bson` fails to build because of the old oasis `setup.ml`. I made a PR to fix it (here : https://github.com/MassD/bson/pull/18), I'll let you know how it goes.

**Comment by dbousque** [comment]
Hi @groar , sorry for the delay. I included `bson` in the sources and published the result in `lymp` 0.2.4. You can update opam (`opam update`), the installation should work fine now. Thanks again for reporting :)
